geomstats/geometry/matrices.py

```diff
@@ -36,13 +39,36 @@ def mul(a, b):
 
         Returns
         -------
-        c : array-like, shape=[n_samples, dim1, dim3]
+        eq : array-like boolean, shape=[n_samples]
         """
-        return gs.matmul(a, b)
+        is_vectorized = \
+            (gs.ndim(gs.array(a)) == 3) or (gs.ndim(gs.array(b)) == 3)
+        axes = (1, 2) if is_vectorized else (0, 1)
+        return gs.all(gs.isclose(a, b, atol=atol), axes)
 
     @staticmethod
-    def commutator(a, b):
-        """Return the commutator of a and b, i.e. :math: `[a, b] = ab - ba`.
+    def mul(*args):
+        """
+        Return the product of matrices a1, ..., an.
+
+        Parameters
+        ----------
+        a1 : array-like, shape=[n_samples, dim_1, dim_2]
+        a2 : array-like, shape=[n_samples, dim_2, dim_3]
+        ...
+        an : array-like, shape=[n_samples, dim_n-1, dim_n]
+
+        Returns
+        -------
+        mul : array-like, shape=[n_samples, dim_1, dim_n]
+        """
+        return reduce(gs.matmul, args)
+
+    @classmethod
+    def commutator(cls, a, b):
```

Comment:
Can we use `mat_a`, `mat_b`? It makes automatic parsing and refactoring easier not to have single-letter variables.

Otherwise the use of `git grep -l 'a'` will return half of the words of the files

Reply:
I also renamed the method to 'bracket' it will be more uniform with stefan's